src/server/internalServer.cpp

```diff
@@ -461,19 +461,27 @@ bool InternalServer::start() {
   sockAddr6.sin6_family = AF_INET6;
   sockAddr6.sin6_port = htons(m_port);
 
-  if (m_addr.empty()) {
-    if (0 != INADDR_ANY) {
-      sockAddr6.sin6_addr = in6addr_any;
-      sockAddr4.sin_addr.s_addr = htonl(INADDR_ANY);
-    }
-    m_addr = kiwix::getBestPublicIp(m_ipMode == IpMode::IPV6 || m_ipMode == IpMode::ALL);
+  if (m_addr.addr.empty() && m_addr.addr6.empty()) { // No ip address provided
+    if (m_ipMode == IpMode::FLEX) m_ipMode = IpMode::ALL;
+    sockAddr6.sin6_addr = in6addr_any;
+    sockAddr4.sin_addr.s_addr = htonl(INADDR_ANY);
+    IpAddress bestIps = kiwix::getBestPublicIps(m_ipMode);
+    if (m_ipMode == IpMode::IPV4 || m_ipMode == IpMode::ALL) m_addr.addr = bestIps.addr;
+    if (m_ipMode == IpMode::IPV6 || m_ipMode == IpMode::ALL) m_addr.addr6 = bestIps.addr6;
   } else {
-    bool ipv6 = inet_pton(AF_INET6, m_addr.c_str(), &(sockAddr6.sin6_addr.s6_addr)) == 1;
-    bool ipv4 = inet_pton(AF_INET, m_addr.c_str(), &(sockAddr4.sin_addr.s_addr)) == 1;
-    if (ipv6){
-       m_ipMode = IpMode::ALL;
-    } else if (!ipv4) {
-      std::cerr << "Ip address " << m_addr << "  is not a valid ip address" << std::endl;
+    bool providedV4 = !m_addr.addr.empty();
+    m_ipMode = providedV4 ? IpMode::IPV4 : IpMode::IPV6; 
+
+    bool validV4 = inet_pton(AF_INET, m_addr.addr.c_str(), &(sockAddr4.sin_addr.s_addr)) == 1;
+    bool validV6 = inet_pton(AF_INET6, m_addr.addr6.c_str(), &(sockAddr6.sin6_addr.s6_addr)) == 1;
```

Comment:
Are these checks made redundant by the introduction of `ipAvailable()`?

Reply:
There is some overlap for sure. 

In the sense of checking if an ip can proceed yes, but not in the sense of providing a better error message (sacrificing a bit of compute time).

For example address `256.0.0.1` would be called invalid, which is more accurate than just saying it is not available.

It really depends on what we value more optimization/validation.

What is your opinion?

Reply:
Here I am concerned about performance least of all. I wanted to find out if we could simplify the code.

But I missed the fact that these are not just checks, they also have a side effect of initializing `sockAddr4.sin_addr.s_addr` and/or `sockAddr6.sin6_addr.s6_addr` so the calls to `inet_pton()` cannot be eliminated. And even if they could - a more adequate error message is a good reason to have those checks in place.

Reply:
The only thing we could do is initialize based on the mode if we would go the other way, but I agree.